### PR TITLE
Multiple backports from 5.19.2

### PR DESCRIPTION
**Pull request by totten**
Especially:

1. `ContactInterchangeTest` and storage-encoding of various fields
2. `ContactApiKeyTest` and handling of joins

### CODE REVIEW

**Comment by totten** [comment]
The clonebh issue is a problem with the test script.

I can confirm that the api4 test suite passes locally on 5.13.7.

On 5.13.6, it mostly passes -- except for `ContactApiKeyTest`. This is because some of security enforcement was consolidated into core (from APIv3|APIv4 to BAO), and 5.13.6 doesn't have that enforcement. This seems to indicate that api4 becomes less secure *if you pair it with an older 5.13.x*. i.e.

| Core version | API4 extension version | Security |
| -- | -- |--|
| <= 4.4.2 | >= 5.13.7 | Less secure |
| <= 4.4.2 | <= 5.13.6 | Less secure |
| >= 4.4.3 | <= 5.13.6 | Least secure |
| >= 4.4.3 | >= 5.13.7 | Most secure |

TLDR: Upgrade them together, or hold them back together. The ordering is not strictly important (it's fine to separate the updates by 30m), but it's not a good idea to leave them mismatched indefinitely.

I'm adding a status check to alert people so that people don't leave it in the transitional combination of 4.4.3+5.13.6.
